src/components/application_manager/include/application_manager/policies/policy_handler.h

```diff
@@ -583,6 +593,19 @@ class PolicyHandler : public PolicyHandlerInterface,
    */
   void OnAppsSearchCompleted(const bool trigger_ptu) OVERRIDE;
 
+  /**
+   * @brief New application was added to application list
+   * @param new_app_id app_id for this application
+   * @param policy_id policy_id for this application
+   */
+  void OnAddedNewApplicationToAppList(const uint32_t new_app_id,
+                                      const std::string policy_id) OVERRIDE;
+
+  /**
+   * @brief Application queue ready for PTU
+   */
+  std::vector<uint32_t> queue_applications_for_ptu_;
```

Comment:
@ydementieiev i think [queue](http://www.cplusplus.com/reference/queue/queue/) will be better container for your purposes

Reply:
@ydementieiev or even [set](http://www.cplusplus.com/reference/set/set/) as you want to keep only unique app id in your container

Reply:
@ydementieiev i think this container might need access synchronization

Reply:
@AKalinich-Luxoft used set and add AutoLock for work with this container in [fe7b393](https://github.com/smartdevicelink/sdl_core/pull/3208/commits/fe7b393fe80f89ac9137d6dd9b41a0dfd6ba28c6)